pennylane/qchem/matrices.py

```diff
@@ -243,23 +263,43 @@ def attraction(*args):
         """
         n = len(basis_functions)
         matrix = qml.math.zeros((n, n))
+
+        requires_grad = getattr(r, "requires_grad", False) or (
+            len(args) > 0 and qml.math.get_interface(r) == "jax" and qml.math.requires_grad(args[0])
+        )
         for (i, a), (j, b) in it.combinations_with_replacement(enumerate(basis_functions), r=2):
             integral = 0
             if args:
                 args_ab = []
-
-                if r.requires_grad:
-                    args_ab.extend([arg[i], arg[j]] for arg in args[1:])
+                if requires_grad:
+                    args_ab.extend(
+                        (
+                            qml.math.array([arg[i], arg[j]], like="jax")
+                            if qml.math.get_interface(arg) == "jax"
+                            else [arg[i], arg[j]]
+                        )
+                        for arg in args[1:]
+                    )
                 else:
-                    args_ab.extend([arg[i], arg[j]] for arg in args)
-
+                    args_ab.extend(
+                        (
+                            qml.math.array([arg[i], arg[j]], like="jax")
+                            if qml.math.get_interface(arg) == "jax"
+                            else [arg[i], arg[j]]
+                        )
+                        for arg in args
+                    )
                 for k, c in enumerate(r):
-                    if c.requires_grad:
+                    if getattr(c, "requires_grad", False) or (
+                        qml.math.get_interface(r) == "jax" and qml.math.requires_grad(args[0])
+                    ):
                         args_ab = [args[0][k]] + args_ab
                     integral = integral - charges[k] * attraction_integral(
                         c, a, b, normalize=False
                     )(*args_ab)
-                    if c.requires_grad:
+                    if getattr(c, "requires_grad", False) or (
+                        qml.math.get_interface(r) == "jax" and qml.math.requires_grad(args[0])
```

Comment:
Can we not simply import `_check_requires_grad` from `.integrals.py` and use it here as well? 😛